### PR TITLE
[PTen]Support XPU for Flatten Kernel

### PR DESCRIPTION
**Pull request by YuanRisheng**
### PR types

Others
### PR changes

OPs
### Describe

Support XPU for Flatten Kernel and Optimize Pten Flatten to be compatible with old Kernel

### CODE REVIEW

File: paddle/fluid/operators/flatten_op.cc [hunk 1]
```diff
@@ -80,13 +80,13 @@ class FlattenOp : public framework::OperatorWithKernel {
     auto input_data_type =
         framework::OperatorWithKernel::IndicateVarDataType(ctx, "X");
 
-    //#ifdef PADDLE_WITH_MKLDNN
+    // #ifdef PADDLE_WITH_MKLDNN
```
Comment (chenwhql) [inline paddle/fluid/operators/flatten_op.cc]: 这些注释能否移除？

File: paddle/pten/kernels/xpu/manipulation.h [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+#include "paddle/pten/core/dense_tensor.h"
+#include "paddle/pten/core/kernel_registry.h"
+
+// See Note [ Why still include the fluid headers? ]
+#include "paddle/fluid/platform/device_context.h"
+
+namespace pten {
+
+using XPUDeviceContext = paddle::platform::XPUDeviceContext;
```
Comment (chenwhql) [inline paddle/pten/kernels/xpu/manipulation.h]: 按照晓伟的设计，这里命名应该没有Device，为XPUContext

File: paddle/pten/kernels/xpu/manipulation.cc [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+// is there a more flexible way to deal with this case?
+template <typename T>
+void FlattenWithXShape(const XPUDeviceContext& dev_ctx,
+                       const DenseTensor& x,
+                       int start_axis,
+                       int stop_axis,
+                       DenseTensor* out,
+                       DenseTensor* xshape) {
+  Flatten<T>(dev_ctx, x, start_axis, stop_axis, out);
+  const auto& in_dims = x.meta().dims;
```
Comment (zyfncg) [inline paddle/pten/kernels/xpu/manipulation.cc]: 推荐使用x.dims()获取dim信息

File: paddle/fluid/operators/flatten_op.cc (hunk 1 above)
Comment (YuanRisheng) [reply paddle/fluid/operators/flatten_op.cc]: done!

File: paddle/pten/kernels/xpu/manipulation.h (hunk 2 above)
Comment (YuanRisheng) [reply paddle/pten/kernels/xpu/manipulation.h]: done!

File: paddle/pten/kernels/xpu/manipulation.cc (hunk 3 above)
Comment (YuanRisheng) [reply paddle/pten/kernels/xpu/manipulation.cc]: done!

**Review by MingMingShangTian — APPROVED** [review]
LGTM

**Review by chenwhql — APPROVED** [review]
LGTM

**Review by zyfncg — APPROVED** [review]

**Review by Shixiaowei02 — APPROVED** [review]

**Pull request merged**